### ISSUE TITLE
Allow simultaneous `path` and `url` under [sources] with `path`-priority

### ISSUE THREAD

**Issue by thchr** [JuliaLang/Pkg.jl#4089]
I have a situation where I'd like to use the [sources] entry of Project.toml, to indicate that a package, X, should first be looked for at some `path` - and, if not present there, be looked for at some `url`. I had hoped this might be possible to do by something like
```toml
[sources]
X = {path="..", url="https://github.com/user/X.jl"}
```
but that's not possible currently, by design, cf. https://github.com/JuliaLang/Pkg.jl/blob/69c6de01909d53b0a4514b11715f829fe0569d98/src/project.jl#L104-L106

I think it would be nice to be able to do this though: it would allow one to develop multiple interdependent private packages locally, preferring local versions specified in `path` if the exist, and then distribute them to collaborators, who might not have a local versions (then pulling from `url`).

Basically, I think it would be natural to let `path` take precedence if both `path` and `url` are present and a directory with a project exists at `path`.
